File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/DTO/DTOUser.java
package abbruzzese.webpage.webpage.server.webpage.DTO;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class DTOUser {

    private String Nombre;

    private String Telefono;
    
    private String Email;

    private String Password;

    private String Matricula;

    public  DTOUser(String Nombre,String Telefono,String Email,String Password, String Matricula){


        this.Nombre=Nombre;
        this.Telefono=Telefono;
        this.Email=Email;
        this.Password=Password;
        this.Matricula=Matricula;
    }

    public  DTOUser(String Nombre,String Email){

        this.Nombre=Nombre;
        this.Email=Email;

    }

    public DTOUser(){}

    public String getNombre() {
        return Nombre;
    }

    public void setNombre(String nombre) {
        Nombre = nombre;
    }

    public String getTelefono() {
        return Telefono;
    }

    public void setTelefono(String telefono) {
        Telefono = telefono;
    }

    public String getEmail() {
        return Email;
    }

    public void setEmail(String email) {
        Email = email;
    }

    public String getPassword() {
        return Password;
    }

    public void setPassword(String password) {
        Password = password;
    }

    public String getMatricula() {
        return Matricula;
    }

    public void setMatricula(String matricula) {
        Matricula = matricula;
    }

}

File: PHP_API/DTO/DTOUser.php
<?php 

namespace App\DTO;

use JsonSerializable;

class DTOUser implements JsonSerializable{

    private $username;
    private $password;
    private $email;
    private $telefono;
    private $matricula;

    // Constructor
    public function __construct($telefono, $username, $password, $email,$matricula) {

        $this->username = $username;
        $this->password = $password;
        $this->email = $email;
        $this->telefono=$telefono;
        $this->matricula=$matricula;
    }

    //Getters y Setters

    public function getUsername(){return $this->username;}
    public function setUsername($username){return $this->username=$username;}

    public function getPassword(){return $this->password;}
    public function setPassword($password){return $this->password=$password;}

    public function getEmail(){return $this->email;}
    public function setEmail($email){return $this->email=$email;}

    public function getTelefono(){return $this->telefono;}
    public function setTelefono($telefono){return $this->telefono=$telefono;}

    public function getMatricula(){return $this->matricula;}
    public function setMatricula($matricula){$this->matricula=$matricula;}

    public function jsonSerialize(): array
    {
        return [
            'telefono' => $this->telefono,
            'username' => $this->username,
            'email' => $this->email,
            'matricula' => $this->matricula,
        ];
    }

}

?>